client/src/components/FrontPage.tsx

```diff
@@ -31,15 +41,42 @@ function FrontPage(): JSX.Element {
         display: 'flex', alignItems: 'center',
         justifyContent: 'space-between', flexDirection: 'row'
       }}>
-        <Typography variant="h1" align="left" sx={{ flexGrow: 1 }}>
-          Your Current Courses
+        <Typography variant="h2" align="left" sx={{ flexGrow: 1 }}>
+          Your Courses
+        </Typography>
+      </Box>
+      {
+        coursesOfUser.length > 0
+          ?
+          <CourseTable data={coursesOfUser} />
+          :
+          <Box sx={{
+            display: 'flex', alignItems: 'left',
+            justifyContent: 'space-between', flexDirection: 'row'
+          }}>
+            <p>
+              You have no courses.
+            </p>
+          </Box>
+      }
+      <Box component="span" sx={{
+        display: 'flex', alignItems: 'center',
+        justifyContent: 'space-between', flexDirection: 'row',
+        mt: 5
+      }}>
+        <Typography variant="h2" align="left" sx={{ flexGrow: 1 }}>
+          Courses
         </Typography>
+        { /* Admins are shown the button for creating a new course */
+          auth.role == SystemRole.Admin &&
+          <Button id='ag_new_course_btn' size='large' variant='contained' onClick={() => {
+            navigate('/create-course');
+          }}>
+            Create New Course
+          </Button>
+        }
       </Box>
-      <BasicGrid data={currentCourses} />
-      <Typography variant="h2" align="left" sx={{ flexGrow: 1, mt: 4 }}>
-        Inactive Courses
-      </Typography>
-      <CourseTable data={previousCourses} />
+      <CourseTable data={courses} />
```

Comment:
Should the "Courses" under "Your courses" render conditionally only for admins?

Now non admin can see empty "Courses" view.

Reply:
I decided to remove `AllCoursesView` and integrate the list to `FrontPage` in this PR because there is very little to differentiate the two views now. But I think it's probably useful in some cases for regular users to see a list of all courses as well (and there's no reason to hide that anyway), so I chose not to conditionally render it.

The list being empty was unintentional, I didn't realize the use of `GET /v1/courses` was locked behind the admin role before.